src/Parser/MineralScanParser.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+namespace RecursiveTree\Seat\TreeLib\Parser;
+
+use Seat\Eveapi\Models\Sde\InvType;
+
+/**
+ * Used to parse mineral scans
+ */
+class MineralScanParser extends Parser
+{
+
+    protected static function parse(string $text, string $EveItemClass)
+    {
+        $expr = implode("", [
+            "^(?<name>[^\t*]+)\*?",           // Name
+            "\t(?<amount>" . self::BIG_NUMBER_REGEXP . "?)", // amount
+            "(?:\t(?<volume>" . self::BIG_NUMBER_REGEXP . ")\sm.)?",         //volume
+            "(?:\t(?<distance>" . self::BIG_NUMBER_REGEXP . "))?\s(?<dist_type>k?m)", // distance
+            "$"
+        ]);
+
+        $lines = self::matchLines($expr, $text);
+
+        if ($lines->where("match", "!=", null)->isEmpty()) return null;
+
+        $warning = false;
+        $parsed = [];
+        $unparsed = [];
```

Comment:
Instead of having $parsed and $unparsed, wouldn't it make more sense to set a property on the EveItemClass?

This way, we also wouldn't need a new FullParseResult class.